cfy_manager/components/prometheus/prometheus.py

```diff
@@ -650,10 +647,6 @@ def _update_manager_alerts_services():
     dest_file_name = join(PROMETHEUS_ALERTS_DIR, 'manager.yml')
     match_pattern = r'name=~"\([a-z\|\-_]*\)'
 
-    with open(src_file_name) as f:
-        prometheus_conf = f.read()
+    prometheus_conf = files.sudo_read(src_file_name)
     new_services = re.findall(match_pattern, prometheus_conf)[0]
-    with open(dest_file_name) as f:
-        prometheus_conf = f.read()
-    with open(dest_file_name, 'wt') as f:
-        f.write(re.sub(match_pattern, new_services, prometheus_conf))
+    files.replace_in_file(match_pattern, new_services, dest_file_name)
```

Comment:
`files.replace_in_file()` does not use `sudo` when reading and that's going to cause problems for `managers.yml`.  You'd better use `sudo_read` and then `write_to_file` (which both use sudo).

Reply:
that's why I change read to sudo_read in replace_in_file below. I think the original was an oversight

Reply:
if it was an oversight, let's fix it then, sure